### ISSUE TITLE
Trailing commas for type parameters in Flow with --trailing-comma=es5

### ISSUE THREAD

**Issue by suchipi** [prettier/prettier#3722]
I'm not sure if this is the correct behavior or not, but I just moved from `trailingComma: "all"` to `trailingComma: "es5"` in our codebase at work, and was surprised to see these ones disappear.

**Prettier 1.9.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAYhCAFQAsBLAZwDkBDAWzgElyA1OAJzQBloBzAZSLFqMBjABCbONQDW5QoLiU4ADxidSUHAF5sA+jDJQeAHlyk25GAAVqbOgBpdcSFAAmNu7QB8AbhD2QCAwYUmhyZFBbNggAdw8EcJRqADcIUld-EGpLZBg2AFc4AIAjOzBpOBg+DGowDR5cgqKQVwgwZAAzagAbcmaNPrZrOx5aak6evoCAK3JlCVqKqro4dU0J3uaIfJgMHYAmDamQGot2ZBBi6mK0buhMjDYNGAB1dINkAA4ABgDHiD6LzsGAujzgg2ScEykgAjvlzHArCMxkdmn1aKRGoUAuR6t04ABFfIQeCogIwa5vVwfJD7cl2UjdeoAYQgtBRKHBAFZMvk+oRrokupsAL4ioA)
```sh
--trailing-comma es5
```

**Input:**
```jsx
type FooThisNameIsVeryLongSoThatItBreaksToTheNextLine = Something<FirstParam, SecondParam>;
```

**Output:**
```jsx
type FooThisNameIsVeryLongSoThatItBreaksToTheNextLine = Something<
  FirstParam,
  SecondParam
>;

```

**Prettier 1.9.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAYhCAFQAsBLAZwDkBDAWzgElyA1OAJzQBloBzAZSLFqMBjABCbONQDW5QoLiU4ADxidSUHAF5sA+jDJQeAHlyk25GAAVqbOgBpdcSFAAmNu7QB8AbhD2QCAwYUmhyZFBbNggAdw8EcJRqADcIUld-EGpLZBg2AFc4AIAjOzBpOBg+DGowDR5cgqKQVwgwZAAzagAbcmaNPrZrOx5aak6evoCAK3JlCVqKqro4dU0J3uaIfJgMHYAmDamQGot2ZBBi6mK0buhMjDYNGAB1dINkAA4ABgDHiD6LzsGAujzgg2ScEykgAjvlzHArCMxkdmn1aKRGoUAuR6t04ABFfIQeCogIwa5vVwfJD7cl2UjdeoAYQgtBRSW63Uy+T6hGuiS6mwAvsKgA)
```sh
--trailing-comma all
```

**Input:**
```jsx
type FooThisNameIsVeryLongSoThatItBreaksToTheNextLine = Something<FirstParam, SecondParam>;
```

**Output:**
```jsx
type FooThisNameIsVeryLongSoThatItBreaksToTheNextLine = Something<
  FirstParam,
  SecondParam,
>;

```

This affects both flow and typescript.

Seeing as type parameters aren't valid ES5 anyway, should we maybe include trailing commas there even when using `trailingComma: "es5"`?

**Comment by suchipi** [comment]
Related: https://github.com/prettier/prettier/issues/3662 https://github.com/prettier/prettier/pull/3313

**Comment by duailibe** [comment]
Following that logic we could consider that `"es5" == "all"` since it's valid in TS anyway, not ES :joy:

**Comment by suchipi** [comment]
Yeah I guess that's kinda weird... I'm looking at it from the flow perspective, though.

**Comment by vjeux** [comment]
@suchipi looks like an oversight to me. Feel free to fix it :)

**Comment by azz** [comment]
From memory, older versions of TS don't support trailing commas here, I think it was intentional. Might want to double check

**Comment by fisker** [comment]
Fixed by #14085
